moveit_ros/moveit_servo/src/servo_calcs.cpp

```diff
@@ -796,22 +798,28 @@ bool ServoCalcs::enforceSRDFPositionLimits()
 // Is handled differently for position vs. velocity control.
 void ServoCalcs::suddenHalt(trajectory_msgs::JointTrajectory& joint_trajectory)
 {
-  if (joint_trajectory.points.empty())
-  {
-    joint_trajectory.points.push_back(trajectory_msgs::JointTrajectoryPoint());
-    joint_trajectory.points[0].positions.resize(num_joints_);
-    joint_trajectory.points[0].velocities.resize(num_joints_);
-  }
-
+  // Prepare the joint trajectory message to stop the robot
+  joint_trajectory.points.clear();
+  joint_trajectory.points.emplace_back();
+  trajectory_msgs::JointTrajectoryPoint& point = joint_trajectory.points.front();
+  point.positions.resize(num_joints_);
+  point.velocities.resize(num_joints_);
+
+  // Assert the following loop is safe to execute
+  assert(point.positions.size() >= num_joints_);
+  assert(point.velocities.size() >= num_joints_);
+  assert(original_joint_state_.position.size() >= num_joints_);
```

Comment:
These will only be compiled in a debug build and will not affect runtime performance in release mode.

Reply:
I'm sorry but testing the specified effect of `vector::resize` does not make any sense! The last assert is probably reasonable, but please remove the others.

I approve already so you can merge after changing this.

Reply:
> I'm sorry but testing the specified effect of `vector::resize` does not make any sense! The last assert is probably reasonable, but please remove the others.

Good point, fixed.  Thank you for reviewing this.